cluster/v1alpha1/types.go

```diff
@@ -301,32 +301,77 @@ const (
 
 // PrioritizerConfig represents the configuration of prioritizer
 type PrioritizerConfig struct {
+	// Name will be deprecated in v1beta1 and replaced by PrioritizerScoreCoordinate.BuildIn.
+	// If both Name and PrioritizerScoreCoordinate.BuildIn are defined, will use the value
+	// in PrioritizerScoreCoordinate.BuildIn.
 	// Name is the name of a prioritizer. Below are the valid names:
 	// 1) Balance: balance the decisions among the clusters.
 	// 2) Steady: ensure the existing decision is stabilized.
 	// 3) ResourceAllocatableCPU & ResourceAllocatableMemory: sort clusters based on the allocatable.
-	// +kubebuilder:validation:Required
-	// +required
-	Name string `json:"name"`
+	// +optional
+	Name string `json:"name,omitempty"`
 
-	// Weight defines the weight of prioritizer. The value must be ranged in [0,10].
+	// PrioritizerScoreCoordinate represents the configuration of the prioritizer and score source.
+	// +optional
+	PrioritizerScoreCoordinate *PrioritizerScoreCoordinate `json:"scoreCoordinate,omitempty"`
+
+	// Weight defines the weight of the prioritizer score. The value must be ranged in [-10,10].
 	// Each prioritizer will calculate an integer score of a cluster in the range of [-100, 100].
 	// The final score of a cluster will be sum(weight * prioritizer_score).
 	// A higher weight indicates that the prioritizer weights more in the cluster selection,
-	// while 0 weight indicate thats the prioritizer is disabled.
-	// +kubebuilder:validation:Minimum:=0
+	// while 0 weight indicates that the prioritizer is disabled. A negative weight indicates
+	// wants to select the last ones.
+	// +kubebuilder:validation:Minimum:=-10
 	// +kubebuilder:validation:Maximum:=10
 	// +kubebuilder:default:=1
 	// +optional
 	Weight int32 `json:"weight,omitempty"`
 }
 
+// PrioritizerScoreCoordinate represents the configuration of the prioritizer and score source
+type PrioritizerScoreCoordinate struct {
+	// Type defines the type of the prioritizer.
+	// Type is either "BuildIn", "AddOn" or "", where "" is "BuildIn" by default.
+	// When the type is "BuildIn", need to specify a buildin prioritizer name in BuildIn.
+	// When the type is "AddOn", need to configure the score source in AddOn.
+	// +kubebuilder:default:=BuildIn
+	// +optional
+	Type string `json:"type,omitempty"`
```

Comment:
should this be required? Would you add enum marker for this

Reply:
Fixed.